src/http/generic_stream_acceptor.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022 The Forgotten Server Authors. All rights reserved.
+// Use of this source code is governed by the GPL-2.0 License that can be found in the LICENSE file.
+
+#include "generic_stream_acceptor.h"
+
+#include <memory>
+
+namespace Http
+{
+
+namespace
+{
+
+namespace detail
+{
+
+template<typename Acceptor, typename Endpoint>
+static void open(Acceptor& acceptor, Endpoint endpoint, ErrorCode& error)
+{
+	acceptor.open(endpoint.protocol(), error);
+	if (error) {
+		return;
+	}
+
+	asio::socket_base::reuse_address option(true);
+	acceptor.set_option(option, error);
+	if (error) {
+		return;
+	}
+
+	acceptor.bind(endpoint, error);
+	if (error) {
+		return;
+	}
+
+	acceptor.listen(asio::socket_base::max_connections, error);
+	if (error) {
+		return;
+	}
+}
+
+} //namespace detail
```

Comment:
This namespace is useless

Reply:
Had to rename `open` template without namespace.